src/inspector_agent.cc

```diff
@@ -486,13 +490,9 @@ void Agent::PostMessages() {
   if (dispatching_messages_)
     return;
   dispatching_messages_ = true;
-  std::vector<blink::protocol::String16> messages;
-  uv_mutex_lock(&queue_lock_);
-  messages.swap(message_queue_);
-  uv_mutex_unlock(&queue_lock_);
-
-  for (auto const& message : messages)
-    inspector_->dispatchMessageFromFrontend(message);
+  for (auto const& message : TakePendingMessages(&message_queue_))
+    inspector_->dispatchMessageFromFrontend(
+        String16(message.c_str(), message.length()));
```

Comment:
Implicit utf8 conversion here, it should be String16::fromUTF8. I wonder at what point we lost it. It would require restoring it upstream.